### ISSUE TITLE
Parity binary download (intermittently) failing in CI

### ISSUE THREAD

**Issue by carver** [ethereum/web3.py#882]
### What was wrong?

It appears that we're (intermittently) getting no data back from the parity binary url json (for example, in #864). See this error:

```
Job specifies Parity version v1.10.4
Traceback (most recent call last):
  File "tests/integration/parity/install_parity.py", line 39, in get_parity_release_json
    params=kwargs).json()
  File "/home/circleci/.local/lib/python3.5/site-packages/requests/models.py", line 892, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/local/lib/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.5/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "tests/integration/parity/install_parity.py", line 113, in <module>
    install_parity(sys.argv[1])
  File "tests/integration/parity/install_parity.py", line 73, in install_parity
    uri, checksum = get_uri(**params)[0]
  File "/home/circleci/.local/lib/python3.5/site-packages/toolz/functoolz.py", line 466, in __call__
    ret = self.first(*args, **kwargs)
  File "/home/circleci/.local/lib/python3.5/site-packages/toolz/functoolz.py", line 440, in memof
    cache[k] = result = func(*args, **kwargs)
  File "tests/integration/parity/install_parity.py", line 44, in get_parity_release_json
    )) from json_err
OSError: Could not find parity release for https://vanity-service.parity.io/parity-binaries with {'version': 'v1.10.4', 'architecture': 'x86_64', 'os': 'debian'}
Exited with code 1
```

### How can it be fixed?

Some options to consider, not mutually exclusive:

- Cache the parity binary much more aggressively (shouldn't need to recreate the cache for the parity binary, basically ever)
- Add some more retries
- Find another source for the binaries, or host it ourselves

**Comment by dylanjw** [comment]
> Cache the parity binary much more aggressively

This was easy, because it seemed the parity binary wasnt getting cached at all.  Add to the circleci cached directories in #885.

**Comment by pipermerriam** [comment]
Fixed in #885 885
